ext/openssl/openssl.c

```diff
@@ -2327,12 +2328,13 @@ PHP_FUNCTION(openssl_x509_read)
 		RETURN_THROWS();
 	}
 	x509 = php_openssl_x509_from_zval(cert, 1, &res);
-	ZVAL_RES(return_value, res);
 
 	if (x509 == NULL) {
-		php_error_docref(NULL, E_WARNING, "Supplied parameter cannot be coerced into an X509 certificate!");
-		RETURN_FALSE;
+		zend_argument_type_error(1, "cannot be coerced into an X509 certificate");
```

Comment:
It's been a while as I was just rebasing, but there are multiple OpenSSL resources (see: https://www.php.net/manual/en/openssl.resources.php) so that message wouldn't be correct either, maybe we should first work on the resource to object conversion and then fave another look at promoting the warnings?

Reply:
Conversion is done :)